conman/routes/handlers.py

```diff
@@ -23,11 +24,48 @@ def check(cls, route):
         return []
 
     def handle(self, request, path):
-        """Raise an error if a subclass calls handle without defining how.."""
+        """Raise an error if a subclass calls handle without defining how."""
         msg = 'Subclasses of `BaseHandler` must implement `handle()`.'
         raise NotImplementedError(msg)
 
 
+class TemplateHandler(BaseHandler):
+    """
+    A handler that renders a template.
+
+    Routes using this handler must define `template_name`, which can either
+    be a string or list, as accepted by `django.shortcuts.render`.
+
+    Renders the `template_name` of the related Route into an
+    `HttpResponse`.
+    """
+    @classmethod
+    def check(cls, route):
+        """Ensure route has a template_name attribute."""
+        if not hasattr(route, 'template_name'):
```

Comment:
Perhaps negate this check and swap the return values?

Reply:
I understand why you'd suggest it -- the bulk of the code resides within the `if`, and "indentation is not your friend". However, in the spirit of eliminating edge cases from continued flow first, I prefer it this way.

Reply:
I feel like in the case of a check, the edge-case is when the check passes.

Reply:
I'm not sure about that.

If we were to add a new unrelated condition that causes failure here, if would simply be a matter of adding another `if` block (and an `errors` list). ie:

```python
errors = []
if failure_condition_1:
    errors.append(error_1)
if failure_condition_2:
    errors.append(error_2)
return []
```

If I were to change this to be the other way, the logic could become complex, because the remaining logic would still need to be conditional:

```python
if not failure_condition_1 and not failure_condition_2:  # Success branch first
    return []
errors = []
if failure_condition_1:
    errors.append(error_1)
if failure_condition_2:
    errors.append(error_2)
return []
```

Reply:
I've been thinking about this a little more, and I think you're right. Apologies.

I believe the flaw in my argument is the idea that a check might have multiple unrelated conditions. If it did, I would better imagine them as two different checks.

I don't think this is an important one to fix at the moment, as I'm keen to get the docs fleshed out, but I'd happily merge the PR if you were to open a correction for this.

Wish I hadn't been quite so hasty to merge now!